### PR TITLE
More API

### PR DESCRIPTION
**Pull request by kianzarrin**
I added a few more API interface definitions that I have used in my mod.
this reduces my reliance on direct calls to inside of TMPE and more future proofing.

in each commit I have added the definitions for one mod.

### CODE REVIEW

**Comment by kianzarrin** [comment]
@krzychu124 Can you please push this to WS sooner than other stuff? Its just definitions so it can't possibly break anything.

File: TLM/TLM/TLM.csproj [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <FileAlignment>512</FileAlignment>
     <TargetFrameworkProfile />
     <LangVersion>preview</LangVersion>
+    <ProduceReferenceAssembly>true</ProduceReferenceAssembly>
```
Comment (krzychu124) [inline TLM/TLM/TLM.csproj:16]: why?

**Review by kvakvs — APPROVED** [review]

File: TLM/TLM/TLM.csproj (hunk 1 above)
Comment (kianzarrin) [reply TLM/TLM/TLM.csproj:16]: @krzychu124 I do need `TrafficManager.Constants` so I need to have a reference to TMPE dll. But that dll is big with all the pictures and all. With this, I can copy a smaller reference assembly to my repo for reference.

Comment (krzychu124) [reply TLM/TLM/TLM.csproj:16]: 4MB is big... whatever.

**Review by krzychu124 — APPROVED** [review]

**Pull request merged**